uint64 col
uint64 row
---
int64 benefit